### ISSUE TITLE
evm: a lot of opcode gadgets need to be updated with `CellType::StoragePhase2`

### ISSUE THREAD

**Issue by lispc** [privacy-scaling-explorations/zkevm-circuits#1086]
eg: gas_price cell should be a phase2 cell.

https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/f7941c1525f3d93153d207d1c33a3abde6624f72/zkevm-circuits/src/evm_circuit/execution/gasprice.rs#L80-L81

**Comment by lispc** [comment]
modify MockProver to catch these types of errors. 

https://github.com/scroll-tech/halo2/pull/25/commits/e409a69746d14e946bd177f640825ea47225e8d8

yeah a lot of tests failed.

**Comment by ed255** [comment]
We would like to address replacing all the word_rlc usage by word hi/lo very soon (probably in the coming weeks).  For this reason we haven't been very thorough with correctness/soundness reviews and analysis related to the usage of the Challange API.  And I think we won't focus too much on fixing these issues ourselves until the word hi/lo is implemented (which will remove a lot of Challenge API usage).  Having said that, any fix related to these issue will be welcome and reviewed!

**Comment by lispc** [comment]
i think hi-lo cannot solve all the challenge API problem? It can solve evm_word and. lookup_input randomness, but will you plan to also remove the dynamic length keccak_input randomness?

We did some experiment earlier but did not continue finishing it 

https://github.com/scroll-tech/zkevm-circuits/pull/216

**Comment by ed255** [comment]
> i think hi-lo cannot solve all the challenge API problem? It can solve evm_word and. lookup_input randomness, but will you plan to also remove the dynamic length keccak_input randomness?

Definitely, word hi-lo will not solve all Challenge API problems, but hopefully it solves many of them (and reduces some circuit complexity).  Our current plan is to keep using the challenge API (via RLC) for the lookups in EVM Circuit and the dynamic length keccak inputs.

> We did some experiment earlier but did not continue finishing it
> 
> [scroll-tech#216](https://github.com/scroll-tech/zkevm-circuits/pull/216)
> 
> btw jonathan made more improvements based on our version
> 
> https://github.com/jonathanpwang/zkevm-circuits/tree/keccak-only-subtree

Interesting! Thanks for sharing these links!
